File: app/src/main/java/pl/piotrskiba/angularowo/models/OffersInfo.java
package pl.piotrskiba.angularowo.models;

import java.util.List;

public class OffersInfo {

    private final int points;
    private final List<AdOffer> adOffers;
    private final List<Offer> offers;

    public OffersInfo(int points, List<AdOffer> adOffers, List<Offer> offers){
        this.points = points;
        this.adOffers = adOffers;
        this.offers = offers;
    }

    public int getPoints() {
        return points;
    }

    public List<AdOffer> getAdOffers() {
        return adOffers;
    }

    public List<Offer> getOffers() {
        return offers;
    }
}

File: app/src/main/java/pl/piotrskiba/angularowo/main/offers/model/OffersInfo.kt
package pl.piotrskiba.angularowo.main.offers.model

import pl.piotrskiba.angularowo.domain.offers.model.OffersInfoModel

data class OffersInfo(
    private val points: Int,
    val pointsLimitReached: Boolean,
    val adOffers: List<AdOffer>,
    val prizeOffers: List<PrizeOffer>,
) {

    val pointsText = points.toString()
}

fun OffersInfoModel.toUi() =
    OffersInfo(
        points,
        pointsLimitReached,
        adOffers.toUi(pointsLimitReached),
        prizeOffers.toUi(points),
    )
